extension/persistence/relational-jdbc/src/main/java/org/apache/polaris/extension/persistence/relational/jdbc/DatasourceOperations.java

```diff
@@ -0,0 +1,184 @@
+          new BufferedReader(
+              new InputStreamReader(
+                  Objects.requireNonNull(classLoader.getResourceAsStream(scriptFilePath)), UTF_8));
+      StringBuilder sqlBuffer = new StringBuilder();
+      String line;
+      while ((line = reader.readLine()) != null) {
+        line = line.trim();
+        if (!line.isEmpty() && !line.startsWith("--")) { // Ignore empty lines and comments
+          sqlBuffer.append(line).append("\n");
+          if (line.endsWith(";")) { // Execute statement when semicolon is found
+            String sql = sqlBuffer.toString().trim();
+            try {
+              int rowsUpdated = statement.executeUpdate(sql);
+              LOGGER.debug("Query {} executed {} rows affected", sql, rowsUpdated);
+            } catch (SQLException e) {
+              LOGGER.error("Error executing query {}", sql, e);
+              // re:throw this as unhandled exception
+              throw new RuntimeException(e);
+            }
+            sqlBuffer.setLength(0); // Clear the buffer for the next statement
+          }
+        }
+      }
+    } catch (IOException e) {
+      LOGGER.error("Error reading the script file", e);
+      throw new RuntimeException(e);
+    } catch (SQLException e) {
+      LOGGER.error("Error executing the script file", e);
+      throw e;
+    }
+  }
+
+  public <T, R> List<R> executeSelect(
+      @Nonnull String query,
+      @Nonnull Class<T> targetClass,
+      @Nonnull Function<T, R> transformer,
+      Predicate<R> entityFilter,
+      int limit)
+      throws SQLException {
+    try (Connection connection = borrowConnection();
+        Statement statement = connection.createStatement();
+        ResultSet s = statement.executeQuery(query)) {
+      return ResultSetToObjectConverter.collect(s, targetClass, transformer, entityFilter, limit);
+    } catch (SQLException e) {
+      LOGGER.error("Error executing query {}", query, e);
+      throw e;
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  public int executeUpdate(String query) throws SQLException {
+    try (Connection connection = borrowConnection();
+        Statement statement = connection.createStatement()) {
+      return statement.executeUpdate(query);
+    } catch (SQLException e) {
+      LOGGER.error("Error executing query {}", query, e);
+      throw e;
+    }
+  }
+
+  public int executeUpdate(String query, Statement statement) throws SQLException {
+    LOGGER.debug("Executing query {} within transaction", query);
+    try {
+      return statement.executeUpdate(query);
+    } catch (SQLException e) {
+      LOGGER.error("Error executing query {}", query, e);
+      throw e;
+    }
+  }
+
+  public void runWithinTransaction(TransactionCallback callback) throws SQLException {
+    Connection connection = null;
+    try {
+      connection = borrowConnection();
+      connection.setAutoCommit(false); // Disable auto-commit to start a transaction
```

Comment:
Does this implementation care about Transaction Isolation levels? Should we have something about that in the README?

Reply:
Presently no, I can take this as an argument, but just for my understanding, do we want to support anything else apart from serialization ? `writeEntities` is only place which needs this.

Reply:
I did not give a lot of though to how current impl. might work with Read Committed isolation (for example), so I posted this question :)

Serializable isolation should work fine, as far as I can tell. 

Making it a config parameter (method arg) and explicitly setting on our JDBC connections sounds reasonable to me.